cfgov/jinja2/v1/_includes/molecules/form-field-with-button.html

```diff
@@ -0,0 +1,50 @@
+{# ==========================================================================
+
+   form-field-with-button.render()
+
+   ==========================================================================
+
+   Description:
+
+   Creates a form field with text when given:
+
+   settings:  An object used to customize the markup.
+
+   settings.btn_text:          The text for the submit button.
+
+   settings.field.required:    Boolean indicating wether the field is
+                               required.
+
+   settings.field.id:          The id for the input.
+
+   settings.field.info:        The text information to display above the field.
+
+   settings.field.label:       The label for the input.
+
+   settings.field.name:        The name for the input.
+
+   settings.field.placeholder: The placeholder used for the input.
+
+   ========================================================================== #}
+
+{% macro render(settings) %}
+<div class="m-form-field-with-button">
+    <div class="form-group">
+        <label for="{{ settings.field.name }}">
+            <b>{{ settings.field.label }}</b>
+            {% if settings.field.required %}
+                (required)
+            {% endif %}
+        </label>
+        <input id="{{ (settings.field.id or settings.field.name) ~
+                   '_' ~ range(1, 100) | random }}"
```

Comment:
As noted in https://github.com/cfpb/cfgov-refresh/pull/1066/files#r43648559, this will in some cases lead to form inputs with the same `id`.